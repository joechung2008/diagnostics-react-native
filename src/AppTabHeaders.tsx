import { useTheme } from '@/ThemeContext';
import { useAppTabHeadersStyles } from '@/hooks/useAppTabHeadersStyles';
import React from 'react';
import { Text, TouchableOpacity, View } from 'react-native';

interface AppTabHeadersProps {
  selectedTab: string;
  setSelectedTab: (tab: string) => void;
  showList: boolean;
  setShowList: (show: boolean) => void;
}

const AppTabHeaders: React.FC<AppTabHeadersProps> = ({
  selectedTab,
  setSelectedTab,
  setShowList,
  showList,
}) => {
  const { colors } = useTheme();
  const dynamicStyles = useAppTabHeadersStyles(colors);

  return (
    <View style={dynamicStyles.tabList}>
      {[
        {
          value: 'extensions',
          label: 'Extensions',
          emoji: showList ? 'ðŸ“‹' : 'ðŸ“Š',
        },
        { value: 'build', label: 'Build Info' },
        { value: 'server', label: 'Server Info' },
      ].map(tab => (
        <TouchableOpacity
          key={tab.value}
          style={[
            dynamicStyles.tab,
            selectedTab === tab.value && dynamicStyles.tabSelected,
          ]}
          onPress={() => {
            if (tab.value === 'extensions' && selectedTab === 'extensions') {
              setShowList(!showList);
            } else {
              setSelectedTab(tab.value);
            }
          }}
        >
          <Text
            style={[
              dynamicStyles.tabText,
              selectedTab === tab.value && dynamicStyles.tabTextSelected,
            ]}
            accessibilityLabel={
              tab.emoji
                ? `${tab.label} ${showList ? 'List view' : 'Details view'}`
                : tab.label
            }
          >
            {tab.label}
            {tab.emoji ? ` ${tab.emoji}` : ''}
          </Text>
        </TouchableOpacity>
      ))}
    </View>
  );
};

export default AppTabHeaders;
