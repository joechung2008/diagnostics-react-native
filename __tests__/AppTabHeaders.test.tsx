import AppTabHeaders from '@/AppTabHeaders';
import { render } from '@testing-library/react-native';
import React from 'react';

// Mock the ThemeContext
jest.mock('@/ThemeContext', () => ({
  useTheme: () => ({
    colors: {
      background: '#ffffff',
      text: '#212529',
      primary: '#007bff',
    },
  }),
}));

// Mock styles hook
jest.mock('@/hooks/useAppTabHeadersStyles', () => ({
  useAppTabHeadersStyles: () => ({
    tabHeader: {},
    tabButton: {},
    activeTab: {},
    inactiveTab: {},
  }),
}));

describe('AppTabHeaders', () => {
  const defaultProps = {
    selectedTab: 'extensions',
    setSelectedTab: jest.fn(),
    showList: true,
    setShowList: jest.fn(),
  };

  it('renders correctly with default props', () => {
    const { toJSON } = render(<AppTabHeaders {...defaultProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with different selected tab', () => {
    const { toJSON } = render(
      <AppTabHeaders {...defaultProps} selectedTab="build" />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when showList is false', () => {
    const { toJSON } = render(
      <AppTabHeaders {...defaultProps} showList={false} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
