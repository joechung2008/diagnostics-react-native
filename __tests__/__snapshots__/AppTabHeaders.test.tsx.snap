// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AppTabHeaders renders correctly when showList is false 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      accessibilityLabel="Extensions Details view"
      style={
        [
          undefined,
          undefined,
        ]
      }
    >
      Extensions
       ðŸ“Š
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      accessibilityLabel="Build Info"
      style={
        [
          undefined,
          false,
        ]
      }
    >
      Build Info
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      accessibilityLabel="Server Info"
      style={
        [
          undefined,
          false,
        ]
      }
    >
      Server Info
    </Text>
  </View>
</View>
`;

exports[`AppTabHeaders renders correctly with default props 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      accessibilityLabel="Extensions List view"
      style={
        [
          undefined,
          undefined,
        ]
      }
    >
      Extensions
       ðŸ“‹
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      accessibilityLabel="Build Info"
      style={
        [
          undefined,
          false,
        ]
      }
    >
      Build Info
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      accessibilityLabel="Server Info"
      style={
        [
          undefined,
          false,
        ]
      }
    >
      Server Info
    </Text>
  </View>
</View>
`;

exports[`AppTabHeaders renders correctly with different selected tab 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      accessibilityLabel="Extensions List view"
      style={
        [
          undefined,
          false,
        ]
      }
    >
      Extensions
       ðŸ“‹
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      accessibilityLabel="Build Info"
      style={
        [
          undefined,
          undefined,
        ]
      }
    >
      Build Info
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      accessibilityLabel="Server Info"
      style={
        [
          undefined,
          false,
        ]
      }
    >
      Server Info
    </Text>
  </View>
</View>
`;
